lib/common/hook.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017, Dell EMC, Inc.
+
+'use strict';
+
+module.exports = hookFactory;
+
+hookFactory.$provide = 'Hook';
```

Comment:
Please rename to "Hooks" as this code is to handle all hooks rather than a particular hook.